File: testapp/java_files/main_palindrome.java
class main_palindrome
{
    public static <E> void check(E expect, E result)
    {
        if(result.equals(expect))
        {
            System.out.println("Output expected "+expect+" and got "+result);
        }
	else
	{
	    System.out.println("Output expected "+expect+" but got "+result);
	    System.exit(1);
	}
    }
    public static void main(String arg[])
    {
       Test t = new Test();
       boolean result;
       result= t.palindrome(123);
       check(false, result);
       result = t.palindrome(151);
       check(true, result);
       result = t.palindrome(23432);
       check(true, result);
    }
}

File: testapp/c_cpp_files/main_palindrome.cpp
#include <stdio.h>
#include <stdlib.h>

extern bool palindrome(int);

template <class T>

void check(T expect, T result)
{
    if (expect == result)
    {
	printf("\nCorrect:\n Expected %d got %d \n",expect,result);
    }
    else 
    {
	printf("\nIncorrect:\n Expected %d got %d \n",expect,result);
	exit (1);
   }
}

int main(void)
{
	bool result;
	result = palindrome(123);
        printf("Input submitted to the function: 123");
	check(false, result);
	result = palindrome(121);
        printf("Input submitted to the function: 121");
	check(true, result);
	printf("All Correct\n");
	return 0;
}
